arangod/Aql/SortExecutor.cpp

```diff
@@ -135,146 +134,129 @@ SortExecutor::~SortExecutor() {
   _infos.getResourceMonitor().decreaseMemoryUsage(_memoryUsageForRowIndexes);
 }
 
-void SortExecutor::initializeInputMatrix(AqlItemBlockInputMatrix& inputMatrix) {
-  TRI_ASSERT(_input == nullptr);
-  ExecutorState state;
-
-  // We need to get data
-  std::tie(state, _input) = inputMatrix.getMatrix();
+void SortExecutor::consumeInput(AqlItemBlockInputRange& inputRange,
+                                ExecutorState& state) {
+  size_t memoryUsageForRowIndexes =
+      inputRange.countDataRows() * sizeof(AqlItemMatrix::RowIndex);
 
-  // If the execution state was not waiting it is guaranteed that we get a
-  // matrix. Maybe empty still
-  TRI_ASSERT(_input != nullptr);
-  if (_input == nullptr) {
-    THROW_ARANGO_EXCEPTION(TRI_ERROR_OUT_OF_MEMORY);
-  }
-  // After allRows the dependency has to be done
-  TRI_ASSERT(state == ExecutorState::DONE);
+  ResourceUsageScope guard(_infos.getResourceMonitor(),
+                           memoryUsageForRowIndexes);
 
```

Comment:
We should use the "number of data rows" information here to reserve memory for `_rowIndexes` here, before we `emplace_back` the rows individually.

Reply:
Addressed